### PR TITLE
Add DCGM Monitor

### PR DESCRIPTION
**Pull request by Tabrizian**
The files below are copied from the official DCGM Python 2 bindings. I made some modifications to the documentation, styling, and errors because of using Python 2 and to be consistent with the style that we are using for our code.

* `dcgm_agent.py`
* `dcgm_field_helpers.py`
* `dcgm_fields.py`
* `dcgm_structs.py`
* `dcgm_value.py`

### CODE REVIEW

File: model_analyzer/monitor/dcgm/dcgm_agent.py
Comment (dzier) [inline model_analyzer/monitor/dcgm/dcgm_agent.py:28]: Empty line before comment

File: model_analyzer/monitor/dcgm/dcgm_agent.py
Comment (dzier) [inline model_analyzer/monitor/dcgm/dcgm_agent.py:33]: Is there a reason these are global methods? Can they be wrapped in a class or namespace? Also, the function name is inconsistent with the rest of the code base. 

1. Method should be `use_underscores_and_lower_case_text`
2. Should have a readable verb, such as `init_dcgm()`, or better yet `dcgm.init()`
3. No method, variable, or class should have a type in their name, i.e. no Hungarian notation.

File: model_analyzer/monitor/dcgm/model.py
Comment (dzier) [inline model_analyzer/monitor/dcgm/model.py:38]: File should be named after class: `dcgm_monitor.py`

**Review by dzier — CHANGES_REQUESTED** [review]

File: model_analyzer/monitor/dcgm/model.py
Comment (xprotobeast2) [inline model_analyzer/monitor/dcgm/model.py:48]: missing parameter hint for `dcgmPath`

File: model_analyzer/monitor/dcgm/model.py
Comment (xprotobeast2) [inline model_analyzer/monitor/dcgm/model.py:133]: Merging [this PR](https://github.com/triton-inference-server/model_analyzer/pull/14) will break this. Not sure how cases like this are generally handled. Would you rebase on my branch, or have mine merged first then rebase? In any case this needs to be changed to fit the new format.

File: model_analyzer/monitor/dcgm/model.py
Comment (xprotobeast2) [inline model_analyzer/monitor/dcgm/model.py:137]: New format, doesn't return record collector but just a list of records.

File: model_analyzer/monitor/dcgm/model.py
Comment (xprotobeast2) [inline model_analyzer/monitor/dcgm/model.py:145]: Here as well

File: model_analyzer/monitor/dcgm/model.py
Comment (xprotobeast2) [inline model_analyzer/monitor/dcgm/model.py:26]: Leave a line here

File: tests/test_dcgm_monitor.py
Comment (xprotobeast2) [inline tests/test_dcgm_monitor.py:26]: Leave a line

File: tests/test_dcgm_monitor.py
Comment (xprotobeast2) [inline tests/test_dcgm_monitor.py:50]: In [this PR](https://github.com/triton-inference-server/model_analyzer/pull/14) I changed the test to check not only the instances but that there were the same number in the resulting list. Let me know if you think we should do something similar here.

**Review by xprotobeast2 — CHANGES_REQUESTED** [review]

File: model_analyzer/monitor/dcgm/model.py
Comment (Tabrizian) [reply model_analyzer/monitor/dcgm/model.py:133]: I can wait until your PR is merged and then rebase.

**Review by xprotobeast2 — APPROVED** [review]

File: model_analyzer/monitor/dcgm/dcgm_monitor.py
Comment (xprotobeast2) [inline model_analyzer/monitor/dcgm/dcgm_monitor.py:102]: I think we're including a short docstring for private functions

Comment (Tabrizian) [reply model_analyzer/monitor/dcgm/dcgm_monitor.py:102]: I added a description in the parent class. I will add it here too.

**Review by dzier — APPROVED** [review]

**Pull request merged**
